### PR TITLE
PrimeGo/solution_2: Cosmetic updates

### PR DESCRIPTION
**Pull request by Rellum**
Added a test and some cosmetic updates to the Go solution. No performance improvements, although I ran benchmarks to ensure no performance regression.

### CODE REVIEW

File: PrimeGo/solution_2/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module prime_solution_2
```
Comment (Rellum) [inline PrimeGo/solution_2/go.mod:1]: Declaring a module makes using the tooling a lot simpler.

File: PrimeGo/solution_2/main.go [hunk 2]
```diff
@@ -1,49 +1,60 @@
...
 var primeCounts = map[uint64]uint64{
-	10:        4,
-	100:       25,
-	1000:      168,
-	10000:     1229,
-	100000:    9592,
-	1000000:   78498,
-	10000000:  664579,
-	100000000: 5761455,
+	10:             4,
```
Comment (Rellum) [inline PrimeGo/solution_2/main.go:14]: I find the thousands separators easier to follow. Happy to revert if I'm in my own camp.

File: PrimeGo/solution_2/main.go [hunk 3]
```diff
@@ -1,49 +1,60 @@
...
+	10_000:         1_229,
+	100_000:        9_592,
+	1_000_000:      78_498,
+	10_000_000:     664_579,
+	100_000_000:    57_61_455,
+	10_000_000_000: 455_052_511,
 }
 
-type Sieve struct {
+type sieve struct {
```
Comment (Rellum) [inline PrimeGo/solution_2/main.go:25]: types shouldn't be exported by default (this is a main package, so we wouldn't be able to reference this type from a different package, anyway).

File: PrimeGo/solution_2/main.go [hunk 4]
```diff
@@ -52,35 +63,48 @@ func (s Sieve) CountPrimes() uint64 {
...
 
 	flag.Uint64Var(&limit, "limit", 1000000, "limit")
 	flag.DurationVar(&duration, "time", 5*time.Second, "duration")
 	flag.Parse()
 
-	stop := make(chan struct{})
 	passes := 0
 	start := time.Now()
-	time.AfterFunc(duration, func() { stop <- struct{}{} })
+	ctx, cancel := context.WithTimeout(context.Background(), duration)
```
Comment (Rellum) [inline PrimeGo/solution_2/main.go:94]: The canonical way of terminating processes using an external signal like the system clock is to use a context variable.

File: PrimeGo/solution_2/main.go [hunk 5]
```diff
@@ -52,35 +63,48 @@ func (s Sieve) CountPrimes() uint64 {
...
+	defer cancel()
 
-loop:
 	for {
-		select {
```
Comment (Rellum) [inline PrimeGo/solution_2/main.go:76]: Removed the channel and select - it isn't necessary when there's no concurrency (although the standard library's context implementation does use a done channel, so this is really just a refactor).

File: PrimeGo/solution_2/main_test.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestCountPrimes(t *testing.T) {
```
Comment (Rellum) [inline PrimeGo/solution_2/main_test.go:10]: A couple of simple unit tests to make future improvements easy to make and review. 

I added a benchmark, too, but haven't committed this as I'm not sure it adds value.

File: PrimeGo/solution_2/main.go [hunk 7]
```diff
@@ -52,35 +63,48 @@ func (s Sieve) CountPrimes() uint64 {
 	return t
 }
 
-func (s Sieve) ValidateResults() bool {
+func (s *sieve) ListPrimes() []uint64 {
```
Comment (Rellum) [inline PrimeGo/solution_2/main.go:66]: Add this mostly for my own benefit, to understand how the implementation works. It outputs the actual prime numbers after the sieve has been executed.

**Comment by rbergen** [comment]
As is by now well documented in earlier PRs, we don't apply cosmetic updates to existing solutions.

**Pull request closed without merging**